doc/flow_counters/routes_flow_counters.md

```diff
@@ -0,0 +1,475 @@
+Example:
+admin@sonic:~$ show flowcnt-route stats
+Route pattern       VRF               Matched routes           Packets          Bytes
+--------------------------------------------------------------------------------------
+3.3.0.0/16          Vrf_default       3.3.1.0/24               100              4543
+                                      3.3.2.3/32               3443             929229
+                                      3.3.0.0/16               0                0
+
+
+show flowcnt-route stats pattern [<prefix-pattern> [ --vrf <vrf>] ]   // show statistics of all routes matching the configured route pattern
+Example:
+admin@sonic:~$ show flowcnt-route stats pattern 3.3.0.0/16
+Route pattern       VRF               Matched routes           Packets          Bytes
+--------------------------------------------------------------------------------------
+3.3.0.0/16          Vrf_default       3.3.1.0/24               100              4543
+                                      3.3.2.3/32               3443             929229
+                                      3.3.0.0/16               0                0
+
+show flowcnt-route stats route [<prefix> [ --vrf <vrf>] ]             // show statistics of the specific route matching the configured route pattern
+Example:
+admin@sonic:~$ show flowcnt-route stats route 3.3.3.2/32 --vrf Vrf_management
+Route                     VRF              Route Pattern           Packets          Bytes
+-----------------------------------------------------------------------------------------
+3.3.3.2/32                Vrf_management   3.3.0.0/16              100              4543
+```
+
+Clear counters:
+
+```
+sonic-clear flowcnt-route    // clear all route flow counters
+Example:
+admin@sonic:~$ sonic-clear flowcnt-route
+Route Flow Counters were successfully cleared
+
+sonic-clear flowcnt-route pattern  [<prefix-pattern> [ --vrf <vrf>] ]   // clear flow counters of all routes matching the configured route pattern
+Example:
+admin@sonic:~$ sonic-clear flowcnt-route pattern 3.3.0.0/16 --vrf Vrf_management
+Flow Counters of all routes matching the configured route pattern were successfully cleared
+
+sonic-clear flowcnt-route route [<prefix> [ --vrf <vrf>] ]  // clear flow counters of the specific route matching the configured prefix
+Example:
+admin@sonic:~$ sonic-clear flowcnt-route route 3.3.3.2/32 --vrf Vrf_management
+Flow Counters of the specified route were successfully cleared
+```
+
+##### YANG model
+
+A new container shall be added to sonic-flex_counter.yang:
+
+```yang
+...
+container FLOW_CNT_ROUTE {
+    /* ROUTE_FLEX_COUNTER_GROUP */
+    leaf FLEX_COUNTER_STATUS {
+        type flex_status;
+    }
+}
+...
+```
+
+#### Config DB Enhancements
+
+A new table FLOW_COUNTER_ROUTE_PATTERN_TABLE shall be added to CONFIG DB.
+
+    ; Defines schema for route flow counter table
+    key                  = "FLOW_COUNTER_ROUTE_PATTERN_TABLE|prefix:vrf"      ; Route pattern (prefix + vrf)
```

Comment:
"FLOW_COUNTER_ROUTE_PATTERN_TABLE|vrf_name|prefix" ?

Reply:
Will follow the existing style used in ROUTE_TABLE:
```
"FLOW_COUNTER_ROUTE_PATTERN_TABLE|vrf_name:prefix"
```

Reply:
That's for APPL_DB, we have different guideline for CONFIG_DB, please use "FLOW_COUNTER_ROUTE_PATTERN_TABLE|vrf_name|prefix"